### ISSUE TITLE
Can't auto-stop training despite validation loss increase over time

### ISSUE THREAD

**Issue by noneedanick** [MIC-DKFZ/nnUNet#829]
I am trying different configurations of network training and I also changed maximum epoch 1000 to 250 for my private task. The problem is whether with default nnUNet options or some little modifications I couldn't overcome overfitting issue on my private task so at every try I am waiting at least extra almost 100 epochs even with maximum 250 epochs. Shouldn't it just cut the process after seeing that the validation loss is not improving ? There is a parameter called "patience", is it related to moving average of validation loss metric (eg. lest say patience = 50, than improvement criteria depends on average of validation loss at last 50 epochs)?

Edit: Also (in addition to every 50 epoch) saving checkpoints even its over fitted.

Thx in advance
![progress](https://user-images.githubusercontent.com/68031733/139602916-d22f8e49-fe9f-4a9e-ab18-3fdabfaa5dd3.png)

**Comment by noneedanick** [comment]
So if I understand correctly, increasing difference between validation and train loss metrics by the time is not that much related to overfitting. Evaluation metric shows us the overall dice score I guess than I will see if it gives me better results after 1000 epochs. Thx Fabian, I am gonna share the results as well in this comment.

**Comment by FabianIsensee** [comment]
The question is what is overfitting to you? If the red line increases then yes this is overfitting as measured by the loss function. But you are not interested in the value of the loss, are you? What you are interested in is the Dice score in the validation set. This is what the green line approximately represents :-)

**Comment by noneedanick** [comment]
Ohhh thx a lot really, this helped a lot for expanding my understanding.
